src/Firebase/Streaming/FirebaseCache.cs

```diff
@@ -153,7 +152,6 @@ public IEnumerable<FirebaseObject<T>> PushData(string path, string data, bool re
                     JsonConvert.PopulateObject(data, obj, this.serializerSettings);
                 }
 
-                this.dictionary[pathElements[0]] = this.dictionary[pathElements[0]];
```

Comment:
This is need to make offline storage work, the dictionary can be an OfflineDatabase and assigning value to "self" triggers an upsert - https://github.com/step-up-labs/firebase-database-dotnet/blob/master/src/Firebase/Offline/OfflineDatabase.cs#L82